agent/runner.go

```diff
@@ -98,6 +99,13 @@ func (r *Runner) Run(ctx context.Context) error {
 	ctx, cancel := context.WithTimeout(ctxmeta, timeout)
 	defer cancel()
 
+	// Add sigterm support for internal context.
+	// Required when the pipeline is terminated by external signals
```

Comment:
it's already registerd for the agent -> cmd/agent/agent.go

Reply:
Dose not work here. Its registered from a background context. See: 
```golang
ctxmeta := metadata.NewOutgoingContext(context.Background(), meta)
```

Reply:
Does one of you know what line 66 does / is needed?